sdk/src/lib.rs

```diff
@@ -1,6 +1,11 @@
+#![cfg_attr(RUSTC_IS_NIGHTLY, feature(specialization))]
```

Comment:
add `cfg(test)` and do similar at other places, too.

Reply:
@mvines Putting messy PR's impl code aside, for deep ABI digesting (deep in the sense of deep-copy), we'll very likely end up being forced to add this magic line in every files needing `freeze_abi`.

I think this is tolerable. Am I right?